### ISSUE TITLE
Doxygen fails to build on 10.14.2 with {graphviz,llvm,qt)

### ISSUE THREAD

**Issue by chipbuster** [Homebrew/homebrew-core#35815]
- [*] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [*] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [*] ran `brew update` and can still reproduce the problem?
- [*] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [*] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Attempting to install Doxygen with graphviz, llvm, and qt

- What happened (include command output)
Build errors out with the following output:
```
==> Downloading http://doxygen.nl/files/doxygen-1.8.15.src.tar.gz
Already downloaded: /Users/chipbuster/Library/Caches/Homebrew/downloads/26299af90764399a4f41a4cd011a73b668c27ef31c7c070f4d3bd65a1869080a--doxygen-1.8.15.src.tar.gz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG
==> make
Last 15 lines from /Users/chipbuster/Library/Logs/Homebrew/doxygen/02.make:
  "initDoxygen()", referenced from:
      _main in main.cpp.o
  "generateOutput()", referenced from:
      _main in main.cpp.o
  "readConfiguration(int, char**)", referenced from:
      _main in main.cpp.o
  "checkConfiguration()", referenced from:
      _main in main.cpp.o
  "adjustConfiguration()", referenced from:
      _main in main.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/doxygen] Error 1
make[1]: *** [src/CMakeFiles/doxygen.dir/all] Error 2
make: *** [all] Error 2
```
[Output of `brew gist-logs doxygen`](https://gist.github.com/chipbuster/a26f2ec35a6403f457950c9a43a589f0)

- What you expected to happen
Successful install.

- Step-by-step reproduction instructions (by running `brew install` commands)
Run `brew install doxygen --with-graphviz --with-qt --with-llvm`

Note: this computer was recently restored from a time machine snapshot after hardware failure, and I've noticed that some of the system settings apparently didn't come along for the ride. I'm not 100% sure that I have everything I need in place, but `brew doctor` doesn't seem to be showing anything wrong.

**Comment by SMillerDev** [comment]
Homebrew will accept patches made by upstream so if you or someone else could make a pull request applying https://github.com/doxygen/doxygen/issues/6750 to the homebrew formula. We can get the fix available to everyone.

**Comment by apjanke** [comment]
The build actually succeeds for me, on 10.13.6.

```
$ brew reinstall -s doxygen
==> Reinstalling doxygen
==> Downloading http://doxygen.nl/files/doxygen-1.8.15.src.tar.gz
######################################################################## 100.0%
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/
==> make
🍺  /usr/local/Cellar/doxygen/1.8.15: 9 files, 14.8MB, built in 2 minutes 26 seconds
```

Perhaps the breakage depends on whether you have MacTeX installed on your machine? I have it on my box, and I think the Homebrew CI boxes have it, so maybe that's how the bottle got built.

**Comment by apjanke** [comment]
Nope, it's not MacTeX: the build succeeds for me as is on my 10.14.2 VM that does not have MacTeX installed. Here's logs from my successful build on 10.14.2: https://gist.github.com/a03188e8d40e5888c351820a74d01ca6.

**Comment by apjanke** [comment]
Anyway, here's a PR to pull in the patch: https://github.com/Homebrew/homebrew-core/pull/35897

I'm still curious to track down why it's only failing on some machines, though.
